src/store/RelayContext.js

```diff
@@ -0,0 +1,269 @@
+  /**
+   * Forces the supplied set of queries to be fetched and written to the store.
+   * Any data that previously satisfied the queries will be overwritten.
+   */
+  forceFetch(
+    querySet: RelayQuerySet,
+    callback: ReadyStateChangeCallback
+  ): Abortable {
+    return this._storeData.getQueryRunner().forceFetch(querySet, callback);
+  }
+
+  /**
+   * Reads query data anchored at the supplied data ID.
+   */
+  read(
+    node: RelayQuery.Node,
+    dataID: DataID,
+    options?: StoreReaderOptions
+  ): ?StoreReaderData {
+    return readRelayQueryData(this._storeData, node, dataID, options).data;
+  }
+
+  /**
+   * Reads query data anchored at the supplied data IDs.
+   */
+  readAll(
+    node: RelayQuery.Node,
+    dataIDs: Array<DataID>,
+    options?: StoreReaderOptions
+  ): Array<?StoreReaderData> {
+    return dataIDs.map(
+      dataID => readRelayQueryData(this._storeData, node, dataID, options).data
+    );
+  }
+
+  /**
+   * Reads query data, where each element in the result array corresponds to a
+   * root call argument. If the root call has no arguments, the result array
+   * will contain exactly one element.
+   */
+  readQuery(
+    root: RelayQuery.Root,
+    options?: StoreReaderOptions
+  ): Array<?StoreReaderData> {
+    const storageKey = root.getStorageKey();
+    const results = [];
+    forEachRootCallArg(root, identifyingArgValue => {
+      let data;
+      const dataID = this._storeData.getQueuedStore()
+        .getDataID(storageKey, identifyingArgValue);
+      if (dataID != null) {
+        data = this.read(root, dataID, options);
+      }
+      results.push(data);
+    });
+    return results;
+  }
+
+  /**
+   * Reads and subscribes to query data anchored at the supplied data ID. The
+   * returned observable emits updates as the data changes over time.
+   */
+  observe(
+    fragment: RelayQuery.Fragment,
+    dataID: DataID
+  ): Observable<?StoreReaderData> {
+    const fragmentPointer = new GraphQLFragmentPointer(
+      fragment.isPlural()? [dataID] : dataID,
+      fragment
+    );
+    return new RelayQueryResultObservable(this._storeData, fragmentPointer);
+  }
+
+  applyUpdate(
+    mutation: RelayMutation,
+    callbacks?: RelayMutationTransactionCommitCallbacks
+  ): RelayMutationTransaction {
+    return this._storeData.getMutationQueue().createTransaction(
+      mutation,
+      callbacks
+    );
+  }
+
+  update(
+    mutation: RelayMutation,
+    callbacks?: RelayMutationTransactionCommitCallbacks
+  ): void {
+    this.applyUpdate(mutation, callbacks).commit();
+  }
+
+  /**
+   * Initializes garbage collection: must be called before any records are
+   * fetched. When records are collected after calls to `scheduleCollection` or
+   * `scheduleCollectionFromNode`, records are collected in steps, with a
+   * maximum of `stepLength` records traversed in a step. Steps are scheduled
+   * via `RelayTaskScheduler`.
+   */
+  initializeGarbageCollection(stepLength: number): void {
```

Comment:
Hmm. Let's keep the `scheduler` implementation in `RelayGarbageCollection` and make this a simple proxy method through to the initialize method on `RelayStoreData`. The signature here can then be `initializeGarbageCollection(scheduler)`.